src/arch-clap.ts

```diff
@@ -0,0 +1,547 @@
+/* eslint-disable no-console, global-require, no-magic-numbers, max-statements */
+import * as Path from "path";
+import * as xsh from "xsh";
+import * as Fs from "fs";
+import * as filterScanDir from "filter-scan-dir";
+import * as xclap from "xclap";
+import * as _ from "lodash";
+
+const typeScriptDevDeps = {
+  // code coverage
+  "@istanbuljs/nyc-config-typescript": "^1.0.1",
+  "source-map-support": "^0.5.16",
+  // types for node.js
+  "@types/node": "^13.7.6",
+  // compilers
+  "ts-node": "^8.6.2",
+  typescript: "^3.8.3"
+};
+
+const eslintDevDeps = {
+  "babel-eslint": "^10.1.0",
+  eslint: "^6.8.0",
+  "eslint-config-walmart": "^2.2.1",
+  "eslint-plugin-filenames": "^1.1.0",
+  "eslint-plugin-jsdoc": "^21.0.0"
+};
+
+const eslintTSDevDeps = {
+  // eslint typescript deps
+  "@typescript-eslint/eslint-plugin": "^2.21.0",
+  "@typescript-eslint/parser": "^2.21.0"
+};
+
+/**
+ * User configurable options for @xarc/module-dev tasks
+ */
+export type XarcModuleDevOptions = {
+  /** force terminal colors in output - *default* `true` */
+  forceColor?: boolean;
+  /** Specify a XClap instance to use - *default* `require("xclap")` */
+  xclap?: any; // type not available for xclap yet
+  /** turn off/on linting tasks (using eslint) - *default* `true` */
+  enableLinting?: boolean;
+  /** Specify typescript config to override the default one */
+  tsConfig?: Record<string, any>;
```

Comment:
I did a bit of digging and it looks like the `typescript` package has a type for the `tsconfig.json` file:

```ts
import { CompilerOptions } from 'typescript';
```

Reply:
it seems CompilerOptions use enum for some values where as the JSON file uses strings.  leaving this as a TODO for future.